src/rtctools_heat_network/workflows/io/write_output.py

```diff
@@ -551,10 +556,77 @@ def _name_to_asset(name):
                         self._asset_fixed_operational_cost_map[asset_name]
                     ][0]
 
-            # Here we add KPIs to the polygon area which allows to visualize them by hoovering over
+                    # Calculate the total energy [Wh] consumed/produced in an area
+                    if asset_name in self.heat_network_components.get("source", []):
+                        try:
+                            total_energy_produced_locally_wh[subarea.name] += np.sum(
+                                results[f"{asset_name}.Heat_source"][1:]
+                                * (self.times()[1:] - self.times()[0:-1])
+                                / 3600.0
+                            )
+                        except KeyError:
+                            total_energy_produced_locally_wh[subarea.name] = np.sum(
+                                results[f"{asset_name}.Heat_source"][1:]
+                                * (self.times()[1:] - self.times()[0:-1])
+                                / 3600.0
+                            )
+                    if asset_name in self.heat_network_components.get("demand", []):
```

Comment:
can you add a comment stating explicit that we include the heat losses ofthe buffer, ates and pipes in the consumed energy?

Reply:
Done